### ISSUE TITLE
[Bug?]: ENOTDIR: not a directory w/ TS 4.4 & PnP

### ISSUE THREAD

**Issue by eltonio450** [yarnpkg/berry#3449]
### Self-service

- [X] I'd be willing to implement a fix

### Describe the bug

Hi there :) !

Summary : When using yarn pnp and typescript 4.4 (watch mode), a weird bug occurs: direct import of files from node_modules are not supported. It may be related to how pnp interacts with typescript for file and module resolution.

the bug is completely described here: https://github.com/microsoft/TypeScript/issues/45891

I am not sure if it's a typescript or yarn bug

the bug is not present when not using pnp

```
/home/antoine/example-tsc/.pnp.cjs:6767
        throw error;
        ^

Error: ENOTDIR: not a directory, stat '/node_modules/mapbox-gl/dist/mapbox-gl.js/package.json'
    at makeError (/home/antoine/example-tsc/.pnp.cjs:5497:24)
    at ENOTDIR (/home/antoine/example-tsc/.pnp.cjs:5518:10)
    at ZipFS.resolveFilename (/home/antoine/example-tsc/.pnp.cjs:7544:25)
    at ZipFS.statSync (/home/antoine/example-tsc/.pnp.cjs:7356:28)
    at /home/antoine/example-tsc/.pnp.cjs:8462:20
    at /home/antoine/example-tsc/.pnp.cjs:9000:58
    at ZipOpenFS.getZipSync (/home/antoine/example-tsc/.pnp.cjs:9145:14)
    at ZipOpenFS.makeCallSync (/home/antoine/example-tsc/.pnp.cjs:9000:17)
    at ZipOpenFS.statSync (/home/antoine/example-tsc/.pnp.cjs:8457:17)
    at CustomStatWatcher.stat (/home/antoine/example-tsc/.pnp.cjs:6759:26) {
  code: 'ENOTDIR'
}
```

Maybe yarn should not throw if the file is not a directory?

### To reproduce

https://github.com/eltonio450/example-tsc

```bash
yarn install
yarn tsc # works
yarn tsc --watch # does not work

### Environment

```shell
System:
    OS: Linux 4.19 Ubuntu 20.04.2 LTS (Focal Fossa)
    CPU: (8) x64 Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
  Binaries:
    Node: 14.16.1 - /tmp/xfs-06a2f8a7/node
    Yarn: 3.0.2 - /tmp/xfs-06a2f8a7/yarn
    npm: 6.14.12 - ~/.nvm/versions/node/v14.16.1/bin/npm
```

**Comment by jakub-g** [comment]
@eltonio450 Thanks for opening the ticket, this should be fixed in #3479

**Comment by eltonio450** [comment]
Thank you !
